oaknational/Oak-Web-Application
# Feat/mux signed videos

## Description

- adds useSignedVideoHook 
- adds getSignedVideoToken 
- adds tests for hook
- adds "swr" library

A playbackPolicy prop is added to VideoPlayer to determine if the token should be fetched or not ("public" | "signed') 

 useSignedVideoHook is called in the VideoPlayer, a token is returned if the playbackPolicy is "signed". The token is passed into the MuxPlayer.  

At the moment all SanityVideos are set to "public" within the SanityVideoPlayer


## Issue(s)

Fixes #1101 

## How to test

To test a signed videos go to a page where there is a video player:
Replace the incoming playbackId and playbackPolicy props with below (no signed video examples currently on the site)

```
  const playbackId = "iVr600df01p002CIj5723gjLVh97iXs3me1bfJ8NlID02Q00";
  const playbackPolicy = "signed";
```

@jack-skerrett-oak looks ok to me but worth checking the videos are working as expected throughout.

1. Go to {cloud link}
2. Click on _______
3. You should see _______

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist
